crates/noirc_frontend/src/monomorphisation/ast.rs

```diff
@@ -26,6 +26,7 @@ pub enum Expression {
     Constrain(Box<Expression>, Location),
     Assign(Assign),
     Semi(Box<Expression>),
+    Log(Box<Expression>),
```

Comment:
We shouldn't need to make a separate expression variant for logging, its syntax is the same as a function call so we can use that. It can be a builtin function so that we wouldn't need a new Operation in ssa either.